lark/lark.py

```diff
@@ -308,28 +309,30 @@ def __init__(self, grammar: 'Union[Grammar, str, IO[str]]', **options) -> None:
                     if self.options.cache is not True:
                         raise ConfigurationError("cache argument must be bool or str")
                         
-                    cache_fn = tempfile.gettempdir() + '/.lark_cache_%s_%s_%s.tmp' % (cache_md5, *sys.version_info[:2])
+                    cache_fn = tempfile.gettempdir() + "/.lark_cache_%s_%s_%s_%s.tmp" % (getpass.getuser(), cache_md5, *sys.version_info[:2])
 
-                if FS.exists(cache_fn):
```

Comment:
Removing the call to `FS.exists` might break the test suite. The `MockFS.open` function does not result in an exception being generated if the file is missing.

Reply:
A call to `file.exists()` and then subsequently `open()` is redundant and susceptible to a race condition.

It looks like the test suite would need reworking to appropriately handle this change. 
Would you like prefer that a test suite fix be added to this PR, or would you prefer the `exists()` check to be re-added?

Reply:
No, what you implemented is pretty much the same I also came to conclude to be the best solution.